config/default/test/contact-summary-extras/get_riskfactors_from_pregnancy.spec.js

```diff
@@ -0,0 +1,81 @@
+const { expect } = require('chai');
+const TestRunner = require('cht-conf-test-harness');
+const harness = new TestRunner();
+const { getRiskFactorsFromPregnancy } = require('../../contact-summary-extras');
+let report = {};
+describe('Tests for getRiskFactorsFromPregnancy() method.', () => {
+  before(() => harness.start());
+  after(async () => { return await harness.stop(); });
+  beforeEach(async () => {
+    await harness.clear();
+    // initialize report
+    report = {
+      form: 'pregnancy',
+      fields: {
+        risk_factors: {
+          risk_factors_history: {
+
+          },
+          r_risk_factor_present: 'yes',
+          risk_factors_present: {
+            primary_condition: '',
+            secondary_condition: ''
+          }
+        }
+      }
+    };
+  });
+  afterEach(() => {
+    expect(harness.consoleErrors).to.be.empty;
+  });
+
+  it(`getRiskFactorsFromPregnancy() returns empty if both primary and secondary risk factors are not present.`, () => {
+    expect(getRiskFactorsFromPregnancy(report)).to.be.empty;
+  });
+
+  it(`getRiskFactorsFromPregnancy() returns empty if the form is not pregnancy form.`, () => {
+    report.fields.risk_factors.risk_factors_present.primary_condition = 'p1';
+    report.form = 'family_planning';
+    expect(getRiskFactorsFromPregnancy(report).length).to.equal(0);
+  });
+
+  it(`getRiskFactorsFromPregnancy() returns primary factors only when secondary factors are empty.`, () => {
+    const primary_conditions = 'p1 p2';
+    report.fields.risk_factors.risk_factors_present.primary_condition = primary_conditions;
+    expect(getRiskFactorsFromPregnancy(report)).to.have.members(['p1', 'p2']);
+  });
+
+  it(`getRiskFactorsFromPregnancy() returns secondary factors only when primary factors are empty.`, () => {
```

Comment:
I think the name of this test contradicts the code change that is proposed.

Reply:
I added tests apart from the issue fix as well and wanted to ensure that the function works as intended, returning all risk factors based on all combination.

Reply:
I don't disagree with the test itself, but the title says `returns secondary factors only when primary factors are empty`. 
Is this still true?

Reply:
Yes. Initially `secondary` tests were being returned only when `primary factors` were empty. This [test](https://github.com/medic/cht-core/pull/8145/files#diff-3fea9cc2cd330e283fada5f9e7dbd6c9f1f15a4ae07c1198f9afbaf45da4ae34R54-R57) ensures that secondary factors are also returned even if primary risk factors exist, actual fix for issue. 

This test and [this](https://github.com/medic/cht-core/pull/8145/files#diff-3fea9cc2cd330e283fada5f9e7dbd6c9f1f15a4ae07c1198f9afbaf45da4ae34R42-R46) test that in one's absence other risk factors are returned.